consensus/fast-sync/src/fast_sync.rs

```diff
@@ -149,6 +207,87 @@ async fn validate_hashes(
     })
 }
 
+async fn validate_block<C>(
+    mut context_svc: C,
+    block: Box<Block>,
+    txs: HashMap<[u8; 32], Transaction>,
+    token: ValidBlockId,
+) -> Result<FastSyncResponse, FastSyncError>
+where
+    C: Service<
+            BlockChainContextRequest,
+            Response = BlockChainContextResponse,
+            Error = tower::BoxError,
+        > + Send
+        + 'static,
+    C::Future: Send + 'static,
+{
+    let BlockChainContextResponse::Context(checked_context) = context_svc
+        .ready()
+        .await?
+        .call(BlockChainContextRequest::GetContext)
+        .await?
+    else {
+        panic!("Context service returned wrong response!");
+    };
+
+    let block_chain_ctx = checked_context.unchecked_blockchain_context().clone();
+
+    let block_hash = block.hash();
+    if block_hash != token.0 {
+        return Err(FastSyncError::BlockHashMismatch);
+    }
+
+    let block_blob = block.serialize();
+
+    let Some(Input::Gen(height)) = block.miner_tx.prefix.inputs.first() else {
+        return Err(FastSyncError::MinerTx(MinerTxError::InputNotOfTypeGen));
+    };
+
+    let txs_vec: Vec<Transaction> = txs.values().cloned().collect();
+    let mut txs = Vec::<VerifiedTransactionInformation>::with_capacity(txs_vec.len());
+    for tx in txs_vec {
+        let data = TransactionVerificationData::new(tx)?;
+        txs.push(VerifiedTransactionInformation {
+            tx_blob: data.tx_blob,
+            tx_weight: data.tx_weight,
+            fee: data.fee,
+            tx_hash: data.tx_hash,
+            tx: data.tx,
+        });
+    }
```

Comment:
This needs to be pulling them in the order they are in the block:

```suggestion
    let mut verified_txs = Vec::with_capacity(txs.len());
    for tx in &block.txs {
        let tx = txs
            .remove(tx)
            .ok_or(FastSyncError::TxsIncludedWithBlockIncorrect)?;

        let data = TransactionVerificationData::new(tx)?;
        verified_txs.push(VerifiedTransactionInformation {
            tx_blob: data.tx_blob,
            tx_weight: data.tx_weight,
            fee: data.fee,
            tx_hash: data.tx_hash,
            tx: data.tx,
        });
    }
```
We will need to add a new new error variant and make `txs` `mut`